src/librustc_trans/back/linker.rs

```diff
@@ -199,6 +200,24 @@ impl<'a> Linker for GnuLinker<'a> {
     fn export_symbols(&mut self, _: &Session, _: &CrateTranslation, _: &Path) {
         // noop, visibility in object files takes care of this
     }
+
+    fn try_gold_linker(&mut self) {
+        let gold_exists = Path::new("/usr/bin/ld.gold").exists();
+        let opt_out = self.sess.opts.cg.disable_gold;
```

Comment:
I’d rather to just have a regular if here: `if self.sess.opts.cg.disable_gold { return; }`